vaadin-crud-flow-parent/vaadin-crud-flow-integration-tests/src/test/java/com/vaadin/flow/component/crud/test/EditorButtonsIT.java

```diff
@@ -0,0 +1,182 @@
+package com.vaadin.flow.component.crud.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebElement;
+
+import com.vaadin.flow.component.button.testbench.ButtonElement;
+import com.vaadin.flow.component.crud.testbench.CrudElement;
+import com.vaadin.flow.component.textfield.testbench.TextFieldElement;
+
+public class EditorButtonsIT extends AbstractParallelTest {
+
+    @Before
+    public void init() {
+        String url = getBaseURL().replace(super.getBaseURL(),
+                super.getBaseURL() + "/vaadin-crud") + "/editorbuttons";
+        getDriver().get(url);
+    }
+
+    @Test
+    public void saveButtonCanBeEnabledWhenFormBecomesDirty() {
+        CrudElement crud = getCrud();
+        crud.getNewItemButton().get().click();
+
+        ButtonElement saveButton = crud.getEditorSaveButton();
+
+        assertFalse("Save button should be disabled on initial load of dialog.",
+                saveButton.isEnabled());
+
+        // make the field dirty
+        TextFieldElement lastNameField = crud.getEditor()
+                .$(TextFieldElement.class).attribute("editor-role", "last-name")
+                .first();
+        lastNameField.setValue("Otto");
+
+        assertTrue("Save button should be enalbed when form becomes dirty.",
+                saveButton.isEnabled());
+    }
+
+    @Test
+    public void saveButtonCanBeDisabledOnTheServer() {
+        CrudElement crud = getCrud();
+
+        crud.getNewItemButton().get().click();
+        assertTrue(crud.isEditorOpen());
+
+        TextFieldElement lastNameField = crud.getEditor()
+                .$(TextFieldElement.class).attribute("editor-role", "last-name")
+                .first();
+        lastNameField.setValue("Otto");
+
+        ButtonElement saveButton = crud.getEditorSaveButton();
+        assertTrue("Save button should be enabled", saveButton.isEnabled());
+        getTestButton("disable-save-button").click();
+        assertFalse("Save button should be disabled", saveButton.isEnabled());
+    }
+
+    @Test
+    public void saveButtonBecomesEnabledWhenFormBecomesDirty() {
```

Comment:
I can not find a difference between this test and this one: https://github.com/vaadin/flow-components/pull/2063/files#diff-842cc27c7969f78d2ecb010b9a9ca07225a93a773e26442a6ed50d8b3fcfc756R27

Am I missing something?

Reply:
Yes they are the same, beside that (I'll remove one of them); Do any of these tests cover the point you mentioned? 

>  I would suggest to also add a test for the default or initial behaviour, where the disabled state of the save button is never modified, and check that the save button gets enabled automatically when the form becomes dirty.
> 
> This test first disables the button, and then enables it again, which might have a different effect depending on the implementation.

Reply:
Yes the two tests here cover the case that I mentioned 👍